net/tcp/tcp_input.c

```diff
@@ -616,6 +616,17 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
       goto drop;
     }
 
+  /* d_appdata should remove the tcp specific option field. */
+
+  if ((tcp->tcpoffset & 0xf0) > 0x50)
+    {
+      len = ((tcp->tcpoffset >> 4) - 5) << 2;
```

Comment:
I do not understand this. The `len` value is unconditionally overwritten few lines below.
Maybe `len = (tcp->tcpoffset >> 4) << 2;` at line 634 should be relocated to the `else` of `if ((tcp->tcpoffset & 0xf0) > 0x50)`?

Reply:
> I do not understand this. The `len` value is unconditionally overwritten few lines below. Maybe `len = (tcp->tcpoffset >> 4) << 2;` at line 634 should be relocated to the `else` of `if ((tcp->tcpoffset & 0xf0) > 0x50)`?

They're calculated separately, so maybe they're a little redundant. Offset do not contain options length when tcp_ipvx_select, so need to skip options field when TCP_NEWDATA flag set. Should I skip options field in tcp_ipvx_select function with IPvxTCP_HDRLEN? It doesn't feel right either.

Reply:
Now changes looks better. I was not saying that your initial code was wrong, but at least hard to track what is going on.